src/test/scala/io/citrine/lolo/bags/CorrelationStudy.scala

```diff
@@ -0,0 +1,440 @@
+import io.citrine.lolo.{PredictionResult, TestUtils}
+import io.citrine.lolo.stats.StatsUtils.makeLinearCorrelatedData
+import io.citrine.lolo.stats.functions.Friedman
+import io.citrine.lolo.trees.multitask.MultiTaskTreeLearner
+import io.citrine.lolo.validation.{Merit, NegativeLogProbabilityDensity2d, StandardConfidence2d}
+import org.apache.commons.math3.random.MersenneTwister
+import org.knowm.xchart.{BitmapEncoder, XYChart}
+import org.knowm.xchart.BitmapEncoder.BitmapFormat
+
+import java.io.{BufferedWriter, File, FileWriter}
+import scala.util.Random
+
+/** A description of the relationship between the two variables for which the correlation estimates are to be studied. */
+sealed trait TestProblem {
+  def name: String
+}
+/** The two variables are generated to have some fixed Pearson correlation coefficient */
+case object Linear extends TestProblem {
+  def name = "linear correlation"
+}
+/** Two variables are generated with a quadratic relationship between them, and then some normally-distributed noise
+  * is applied to one variable. The larger the level of noise, the less correlated the two variables are.
+  */
+case object Quadratic extends TestProblem {
+  def name = "quadratic relationship"
+}
+
+/** Metric used to evaluate the accuracy of the estimated covariance matrix. */
+sealed trait Metric {
+  def name: String
+}
+case object NLPD extends Metric {
+  def name = "negative log probability density"
+}
+case object StdConfidence extends Metric {
+  def name = "1-sigma standard confidence"
+}
+
+/** Underlying function from which to draw training and test data. */
+sealed trait TrueFunction {
+  def function: Seq[Double] => Double
+  val numCols: Int
+  def name: String
+}
+case class FriedmanSilvermanFunction(numCols: Int) extends TrueFunction {
+  def name = s"Friedman-Silverman $numCols columns"
+  def function: Seq[Double] => Double = Friedman.friedmanSilverman
+}
+case class FriedmanGrosseSilvermanFunction(numCols: Int) extends TrueFunction {
+  def name = s"Friedman-Grosse-Silverman $numCols columns"
+  def function: Seq[Double] => Double = Friedman.friedmanGrosseSilverman
+}
+
+/** Parameter to vary as part of a scan. */
+sealed trait VariedParameter {
+  def name: String
+}
+case object TrainRho extends VariedParameter {
+  def name: String = "linear training correlation"
+}
+case object TrainQuadraticFuzz extends VariedParameter {
+  def name: String = "shift to decorrelate quadratic data"
+}
+case object Noise extends VariedParameter {
+  def name: String = "observational noise level"
+}
+/** If the number of bags is not being varied, then the default is to set numBags = numTrain */
+case object Bags extends VariedParameter {
+  def name: String = "number of bags"
+}
+case object NumTraining extends VariedParameter {
+  def name: String = "number of training rows"
+}
+
+object CorrelationStudy {
+
+  def main(args: Array[String]): Unit = {
+    val mainRng = new Random(52109317L)
+
+    runTrialsAndSave(
```

Comment:
This is a single example. Ultimately, this `main` method will be a series of `runTrialsAndSave` calls, one for each scan we want to perform. Each such call produces a chart and a csv